packages/flutter/lib/src/material/snack_bar_theme.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Configures whether [SnackBar] should be fixed to the bottom or floating
+/// above other widgets fixed to the bottom, such as [FloatingActionButton].
+enum SnackBarBehavior {
+  /// Fixed the [SnackBar] position to the bottom of the [Scaffold] when
+  /// possible. One of the possible scenarios where [SnackBar] will be shown
+  /// above another widget is above a [BottomNavigationBar]. Other than that
+  /// [SnackBar] will cause other non-fixed widgets inside [Scaffold] to be
+  /// pushed above (e.g. [FloatingActionButton]).
+  fixed,
+
+  /// Change the design and behavior of [SnackBar] to float as described in
+  /// <https://material.io/design/components/snackbars.html>.
+  /// This behavior will cause [SnackBar] to be shown on top of other non-fixed
+  /// widgets, such as [FloatingActionButton], rather than pushing it above
+  /// [SnackBar].
+  floating
+}
+
+/// Defines default property values for [SnackBar] widgets.
+///
+/// Descendant widgets obtain the current [SnackBarThemeData] object using
+/// `Theme.of(context).snackBarTheme`. Instances of [SnackBarThemeData] can be
+/// customized with [SnackBarThemeData.copyWith].
+///
+/// Typically a [SnackBarThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.snackBarTheme].
+///
+/// All [SnackBarThemeData] properties are `null` by default. When null, the
+/// [SnackBar] will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class SnackBarThemeData extends Diagnosticable {
+
+  /// Creates a theme that can be used for [ThemeData.snackBarTheme].
+  ///
+  /// The [elevation] must be null or non-negative.
+  const SnackBarThemeData({
+    this.backgroundColor,
+    this.actionTextColor,
+    this.disabledActionTextColor,
+    this.elevation,
+    this.shape,
+    this.behavior = SnackBarBehavior.fixed,
```

Comment:
Ah it looks like you do have a default check in the snackbar class it just won't be triggered in the default case because `snackBarTheme.behavior` will not be `null`.

Reply:
That makes sense, I've removed the default value for behavior in the theme since the widget itself already has `fixed` as the fallback